src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerSmall.java

```diff
@@ -0,0 +1,190 @@
+        final long mask = (1L << kSize*2) - 1L;
+        // number of unused bits at the top
+        final int compK = 64 - kSize*2;
+        // move the significant bits up to the top, reverse complement, and mask to kSize bits.
+        return new SVKmerSmall(reverseComplement(valLow << compK) & mask);
+    }
+
+    /**
+     * Returns a SVKmerSmall that is a canonical representation of this one.
+     * An odd-K SVKmerSmall is in canonical form if its middle base is A or C.
+     * The reverse-complement of a non-canonical SVKmerSmall is a canonical SVKmerSmall, and vice versa.  (Think about it.)
+     * Canonical form is not defined for even-K Kmers (too expensive to compute routinely).
+     */
+    public SVKmerSmall canonical( final int kSize ) {
+        Utils.validateArg( (kSize & 1) != 0, "K must be odd to canonicalize.");
+        // for odd-size kmers, the high bit of the middle base is in least significant position in valHigh.
+        // test its value by ANDing with 1.  if it's zero the middle base is A or C and we're good to go.
+        if ( ((valLow >> kSize) & 1L) == 0 ) return this;
+        // middle base is G or T.  reverse complement.
+        return reverseComplement(kSize);
+    }
+
+    public final Base firstBase( final int kSize ) { return Base.values()[(int)(valLow >> (kSize*2-2))]; }
+    public final Base lastBase() { return Base.values()[(int)(valLow & 3)]; }
+
+    @Override
+    public boolean equals( final Object obj ) {
+        return obj instanceof SVKmerSmall && equals((SVKmerSmall)obj);
+    }
+
+    public final boolean equals( final SVKmerSmall that ) {
+        return this.valLow == that.valLow;
+    }
+
+    @Override
+    public final int hashCode() {
+        // 32-bit FNV-1a algorithm
+        return fnvLong((int)2166136261L, valLow);
+    }
+
+    /**
+     * SVKmerSmall comparison is consistent with equals.
+     * It's also the same as the lexicographic ordering you'd get using toString on the Kmers.
+     */
+    @Override
+    public final int compareTo( final SVKmerSmall that ) {
+        return Long.compare(this.valLow, that.valLow);
+    }
+
+    /**
+     * Not an override.  An SVKmerSmall doesn't know what K is, so it has to be supplied.
+     */
+    public final String toString( final int kSize ) {
+        final StringBuilder sb = new StringBuilder(kSize);
+
+        // we'll produce the string in reverse order and reverse it at the end
+        long val = valLow;
+        for ( int nChars = kSize/2; nChars > 0; --nChars ) {
+            // grab the two least significant bits to index into the BASE_CHARS array
+            sb.append(BaseUtils.BASE_CHARS[(int)val & 3]);
+            // roll the whole mess down two bits
+            val >>= 2;
+        }
+        // this for loop will have one more iteration than the previous one if kSize is odd
+        for ( int nChars = (kSize+1)/2; nChars > 0; --nChars ) {
```

Comment:
this loop was necessary when there were two longs, but it's superfluous in this class.  correction above does it all in the first for loop.